hawc/apps/animalv2/models.py

```diff
@@ -0,0 +1,576 @@
+    )
+    name = models.CharField(
+        max_length=80,
+        verbose_name="Treatment name",
+        help_text="""
+            TODO
+            """,
+    )
+    chemical = models.ForeignKey(Chemical, on_delete=models.CASCADE, related_name="v2_treatments")
+    route_of_exposure = models.CharField(
+        max_length=2,
+        choices=constants.RouteExposure.choices,
+        help_text="Primary route of exposure. If multiple primary-exposures, describe in notes-field below",
+    )
+    exposure_duration = models.FloatField(
+        verbose_name="Exposure duration (days)",
+        help_text="Length of exposure period (fractions allowed), used for sorting in visualizations. For single-dose or multiple-dose/same day gavage studies, 1.",
+        blank=True,
+        null=True,
+    )
+    exposure_duration_description = models.CharField(
+        verbose_name="Exposure duration (text)",
+        max_length=128,
+        blank=True,
+        help_text="Length of time between start of exposure and outcome assessment, "
+        "in days when &lt;7 (e.g., 5d), weeks when &ge;7 days to 12 weeks (e.g., "
+        "1wk, 12wk), or months when &gt;12 weeks (e.g., 15mon). For repeated "
+        'measures use descriptions such as "1, 2 and 3 wk".  For inhalations '
+        'studies, also include hours per day and days per week, e.g., "13wk '
+        '(6h/d, 7d/wk)." This field is commonly used in visualizations, so '
+        "use abbreviations (h, d, wk, mon, y) and no spaces between numbers "
+        "to save space. For reproductive and developmental studies, where "
+        'possible instead include abbreviated age descriptions such as "GD1-10" '
+        'or "GD2-PND10". For gavage studies, include the number of doses, e.g. '
+        '"1wk (1dose/d, 5d/wk)" or "2doses" for a single-day experiment.',
+    )
+    exposure_outcome_durection = models.FloatField(
+        verbose_name="Exposure-outcome duration (days)",
+        help_text="Optional: Numeric length of time between start of exposure and outcome assessment in days. "
+        + "This field may be used to sort studies which is why days are used as a common metric.",
+        blank=True,
+        null=True,
+    )
+    comments = models.TextField(blank=True)
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.name
+
+    def get_assessment(self):
+        return self.experiment.get_assessment()
+
+    def get_study(self):
+        return self.experiment.get_study()
+
+    # TODO - flat_complete_header_row / flat_complete_data_row
+
+
+class DoseGroup(models.Model):
+    objects = managers.DoseGroupManager()
+
+    # TODO
+    TEXT_CLEANUP_FIELDS = ()
+
+    treatment = models.ForeignKey(
+        Treatment, on_delete=models.CASCADE, related_name="v2_dose_groups"
+    )
+    dose = models.FloatField(validators=[MinValueValidator(0)])
```

Comment:
i think we'll need some sort of dose_index here like our old animal model so it's easy to compare that these are the same groups, regardless of the specified dose units. We could also flip to something like this:


```python
class DoseGroup:
    treatment = ForeignKey
    dose_index = PositiveSmallIntegerField()

class DoseRepresentation:
    group = ForeignKey(DoseGroup)
    dose = FloatField()
    units = ForeignKey(DoseUnits)
```

But it may be better to keep it a single model instead of breaking it up...

Reply:
@shapiromatron is animalv1::DoseGroup::dose_group_id the index you mean?

Reply:
confirmed on phone, "yes" that's the one

Reply:
Yes the `dose_group_id` is the index I meant. Sorry for not being clear earlier!